x-pack/plugins/ml/public/settings/calendar/edit/import_modal.js

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import React, {
+  Component,
+  Fragment
+} from 'react';
+import { PropTypes } from 'prop-types';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiCheckbox,
+  EuiFilePicker,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiSpacer
+} from '@elastic/eui';
+import moment from 'moment';
+
+import { EventsTable } from './events_table';
+
+const icalendar = require('icalendar');
+
+const MAX_FILE_SIZE_MB = 100;
+
+export class ImportModal extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      includePastEvents: false,
+      allImportedEvents: [],
+      fileLoading: false,
+      fileLoaded: false,
+      errorMessage: null,
+    };
+  }
+
+  selectedEvents = [];
+
+  static createEvents = (ical) => {
```

Comment:
Glad you brought it up since this is one I wasn't sure whether to keep this functionality here or move it. 🤔  Mostly kept it because the functionality is all very related and specific to this component and I thought it might be better having them all in one place.

Happy to move to another file - maybe a utils for this component - if it feels like the better way.

Reply:
Decided to move this out of the file to improve class readability.